inginious/frontend/pages/course_user_settings.py

```diff
@@ -0,0 +1,104 @@
+
+        return self.show_page(courseid, course_user_settings, ("success","Course settings successfully updated."))
+
+    def show_page(self, courseid, course_user_settings, feedback):
+        """
+            Definition of the show page method.
+            :param: courseid: the id of the course.
+            :param: course_user_settings: The dict with the settings values.
+            :param: feedback: a tuple with the type of feedback and the feedback. None if there is no feedback.
+            :return:
+        """
+        try:
+            course = self.course_factory.get_course(courseid)
+            course_user_setting_fields = course.get_course_user_settings()
+            return self.template_helper.render("course_user_settings.html", course=course,
+                                               course_user_setting_fields=course_user_setting_fields,
+                                               course_user_settings=course_user_settings, fieldtypes=FieldTypes,
+                                               feedback=feedback)
+        except Exception:
+            return self.template_helper.render("course_unavailable.html")
+
+    def _is_accessible(self, courseid, username):
+        """
+            Verify if course is accessible and that courseid match an existing course.
+            :param: courseid - The id of the course.
+            :param: username - The username of the logged user.
+            :return: A boolean if the course is accessible for the given user.
+        """
+        try:
+            course = self.course_factory.get_course(courseid)
+            return self.user_manager.course_is_user_registered(course, username)
+        except Exception:
+            return False
+
+    def _sanitize_content(self, data, courseid):
+        """
+            Sanitize received data
+            :param: data  - A dictionary that normally contains user settings for a course.
+            :param: courseid - Id of the course.
+            :return: A sanitized dict or a str feedback if something went wrong.
+        """
+        if not isinstance(data, dict):
+            return "Incorrect type of data."
+        copied_data = {}
+        course = self.course_factory.get_course(courseid)
+        add_fields = course.get_course_user_settings()
+        for setting in data:
+            if setting not in add_fields:
+                # if setting is not expected in the course definition. There is no reason to treat it.
+                return "Unknown field: "+str(setting)
+            try:
+                # try to cast given value to be sure that we match expected type.
+                if add_fields[setting].get_type_name() == "STRING":
+                    casted_value= str(data[setting])
+                elif add_fields[setting].get_type_name() == "INTEGER":
+                    casted_value = int(data[setting])
```

Comment:
It's probably useful to allow empty setting, or to define default values for each types.

Indeed, values are not set until student specify them. Therefore, you can send values to the grading environment that wouldn't be accepted by your form, leaving the course administrator to deal with invalid values.

Reply:
There is still an error returned in case you leave an integer field empty.